python/pyiceberg/schema.py

```diff
@@ -15,6 +15,8 @@
 # specific language governing permissions and limitations
 # under the License.
 # pylint: disable=W0511
+from __future__ import annotations
```

Comment:
@Fokko, what is our policy on using this vs using strings? How do we choose?

Reply:
I can add that to [the `Type annotations` section](https://py.iceberg.apache.org/contributing/#type-annotations). I prefer the `from __future__ import annotations` since it introduces less visual noise.